wasm.md

```diff
@@ -600,26 +601,58 @@ Here, we allow for an "abstract" function declaration using syntax `func_::___`,
 
     syntax FuncDecl  ::= "(" FuncDecl ")"     [bracket]
                        | TypeKeyWord ValTypes
-                       | "export" Identifier
     syntax FuncDecls ::= List{FuncDecl, ""} [klabel(listFuncDecl)]
  // --------------------------------------------------------------
 
-    syntax Instr ::= "(" "func"              FuncDecls Instrs ")"
-                   | "(" "func" Identifier FuncDecls Instrs ")"
-                   | "func" Identifier "::" FuncType VecType "{" Instrs "}"
- // -----------------------------------------------------------------------
-    rule <k> ( func FDECLS INSTRS )
-          => func gatherExportedName(FDECLS) :: gatherFuncType(FDECLS) gatherTypes(local, FDECLS) { INSTRS }
+
+    syntax FuncAbbr ::= "(" "export" String ")" FuncAbbr // `import`, if exists, should always be the last one. So should add subsort `| "(" "import" Identifier ")"` in the future.
+                      | "(" "export" String ")"
+    syntax Instr ::= #handleExports ( FuncAbbr )
+ // --------------------------------------------
+    rule <k> #handleExports ( ( export ENAME ) FABBR:FuncAbbr )
+          => ( export ENAME ( func NEXTIDX ) )
+          ~> #handleExports( FABBR )
+          ...
+         </k>
+         <nextFuncIdx> NEXTIDX </nextFuncIdx>
+    rule <k> #handleExports ( ( export ENAME ) )
+          => ( export ENAME ( func NEXTIDX ) )
+          ...
+         </k>
+         <nextFuncIdx> NEXTIDX </nextFuncIdx>
+
+    syntax Instr ::= "(" "func"                     FuncDecls Instrs ")"
+                   | "(" "func"            FuncAbbr FuncDecls Instrs ")"
+                   | "(" "func" Identifier          FuncDecls Instrs ")"
+                   | "(" "func" Identifier FuncAbbr FuncDecls Instrs ")"
+                   | "func"               FuncType VecType "{" Instrs "}"
+                   | "func" Identifier "::" FuncType VecType "{" Instrs "}" // this function is only used in some legacy test cases.
+ // --------------------------------------------------------------------------------------------------------------------------------
```

Comment:
Remove the case which is only used for our legacy tests and update the tests.

Reply:
fixed